app/src/main/java/com/example/concordia_campus_guide/Fragments/PathInfoCardFragment/PathInfoCardViewModel.java

```diff
@@ -5,56 +5,39 @@
 import com.example.concordia_campus_guide.Adapters.DirectionWrapper;
 import com.example.concordia_campus_guide.ClassConstants;
 import com.example.concordia_campus_guide.Models.Direction;
+import com.example.concordia_campus_guide.R;
 import com.google.android.gms.maps.model.LatLng;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class PathInfoCardViewModel extends ViewModel {
-    private double totalDuration;
-
-    public PathInfoCardViewModel(){
-        // used in tests
-    }
-
-    public double getTotalDuration(){
-        return this.totalDuration;
-    }
-
-    public double deg2rad(double deg) {
-        return deg * (Math.PI / 180);
-    }
-
-    public double getDistanceFromLatLonInKm(double lat1, double lon1, double lat2, double lon2) {
-        double earthRadius = 6371; // Radius of the earth in km
-        double dLat = deg2rad(lat2 - lat1);  // deg2rad below
-        double dLon = deg2rad(lon2 - lon1);
-        double angle =
-                Math.sin(dLat / 2) * Math.sin(dLat / 2) +
-                        Math.cos(deg2rad(lat1)) * Math.cos(deg2rad(lat2)) *
-                                Math.sin(dLon / 2) * Math.sin(dLon / 2);
-        double center = 2 * Math.atan2(Math.sqrt(angle), Math.sqrt(1 - angle));
-        return earthRadius * center; // Distance in km
-    }
-
-    public List<Direction> createOutdoorDirectionsList(List<DirectionWrapper> directionsResults){
-        List<Direction> directionList = new ArrayList<>();
-        totalDuration = 0;
-        for (DirectionWrapper directionWrapper : directionsResults) {
-            Direction direction = directionWrapper.getDirection();
-            double minute = direction.getDuration() / 60;
-            direction.setDuration(Math.ceil(minute));
-            totalDuration += minute;
-
-            directionList.add(direction);
+    public PathInfoCardViewModel(){}
+
+    public int getIcon(String type) {
+        switch (type) {
+            case "ELEVATOR":
+                return R.drawable.ic_elevator_white;
+            case "CLASSROOM":
+            case "WALKING":
+                return R.drawable.ic_directions_walk_red;
+            case "ENTRANCE":
+                return R.drawable.ic_exit_to_app_24px_1;
+            case "STAFF_ELEVATOR":
+                return R.drawable.ic_staff_elevator_white;
+            case "STAIRS":
+                return R.drawable.ic_stairs_white;
+            case "TRANSIT":
+                return R.drawable.ic_directions_bus_black_24dp;
+            case "DRIVING":
+                return R.drawable.ic_directions_car_red;
+            case "BICYCLING":
+                return R.drawable.ic_directions_bike_black_24dp;
+            case "SHUTTLE":
+                return R.drawable.ic_shuttle_red;
+            default:
+                return -1;
```

Comment:
hmmm is there a way to avoid this ? because it will increase the cyclomatic complexity . Do u guys think you can store the icons in the assets instead ? you can access them by basically looking up for the file name instead

Reply:
agreed, u will then be able to access them more dynamically based on the name. so like an icon for walking will be called  drection_walking...